spec.md

```diff
@@ -848,7 +848,7 @@ The CO MUST implement the specified error recovery behavior when it encounters t
 | Volume published but is incompatible | 6 ALREADY_EXISTS | Indicates that a volume corresponding to the specified `volume_id` has already been published at the node corresponding to the specified `volume_id` but is incompatible with the specified `volume_capability` or `readonly` flag . | Caller MUST fix the arguments before retying. |
 | Volume published to another node | 9 FAILED_PRECONDITION | Indicates that a volume corresponding to the specified `volume_id` has already been published at another node and does not have MULTI_NODE volume capability. If this error code is returned, the Plugin SHOULD specify the `node_id` of the node at which the volume is published as part of the gRPC `status.message`. | Caller SHOULD ensure the specified volume is not published at any other node before retrying with exponential back off. |
 | Max volumes attached | 8 RESOURCE_EXHAUSTED | Indicates that the maximum supported number of volumes that can be attached to the specified node are already attached. Therefore, this operation will fail until at least one of the existing attached volumes is detached from the node. | Caller MUST ensure that the number of volumes already attached to the node is less then the maximum supported number of volumes before retrying with exponential backoff. |
-| Operation pending for volume | 10 ABORTED | Indicates that there is a already an operation pending for the specified volume. In general the Cluster Orchestrator (CO) is responsible for ensuring that there is no more than one call "in-flight" per volume at a given time. However, in some circumstances, the CO MAY lose state (for example when the CO crashes and restarts), and MAY issue multiple calls simultaneously for the same volume. The Plugin, SHOULD handle this as gracefully as possible, and MAY return this error code to reject secondary calls. | Caller SHOULD ensure that there are no other calls pending for the specified volume, and then retry with exponential back off. |
+| Operation pending for volume | 10 ABORTED | Indicates that there is a already an operation pending for a different RPC type for the specified volume. In general the Cluster Orchestrator (CO) is responsible for ensuring that there is no more than one call "in-flight" per volume at a given time. However, in some circumstances, the CO MAY lose state (for example when the CO crashes and restarts), and MAY issue multiple calls simultaneously for the same volume. The Plugin, SHOULD handle this as gracefully as possible, and MAY return this error code to reject secondary calls. | Caller SHOULD ensure that there are no other calls pending for the specified volume, and then retry with exponential back off. |
```

Comment:
Here it is not clear whether this error is really only about RPCs of a different type.  What about multiple controller publish calls for different nodes, in cases where that is allowed?  

If we are really standing behind the statement that the CO should ensure that `there is no more than one call "in-flight" per volume at a given time` then concurrent ControllerPublishVolume calls could also return `ABORTED` here.

Reply:
Thats a really good point. However, I believe the current wording does allow for multiple controller publish calls for different nodes since that is not a `different RPC type`. It would just require a response of `ABORTED` if say you called `DeleteVolume` while a `ControllerPublishVolume` was still pending.

And I don't think `there is no more than one call "in-flight" per volume at a given time` is a rule at all - it seems to me more like a description of what kind of behavior we may expect or strive for, as its prefaced by `In general` and followed by examples of how it may not be true.